examples/qvgviewer/CMakeLists.txt

```diff
@@ -0,0 +1,45 @@
+list(APPEND TARGET_HEADERS
+    MainWindow.h)
+
+list(APPEND TARGET_SOURCES
+    MainWindow.cpp
+    main.cpp)
+
+set(TARGET_NAME qvgviewer)
+
+#CONFIG += autoqvg
+
+option(autoqvg "enable automatic qvg commpilation" OFF)
+
+if(autoqvg)
```

Comment:
@uwerat what's the idea behind this if statement? 

Should qvg files really be generated into the source directory? I would suggest creating a custom target / command that generates the qvg files from svg into e.g. CMAKE_CURRENT_BINARY_DIR and add them as sources to the target. 

Note: This process would only run once (or if the svg changes)

Reply:
```cmake
if(TARGET Qt5::Svg)
    function(compile_svg_to_qvg SVG_FILENAME QVG_FILENAME)
        get_filename_component(SVG_FILENAME ${SVG_FILENAME} ABSOLUTE)
        add_custom_command(
            COMMAND svg2qvg ${SVG_FILENAME} ${QVG_FILENAME}
            OUTPUT ${QVG_FILENAME}
            DEPENDS ${SVG_FILENAME}
            WORKING_DIRECTORY $<TARGET_FILE_DIR:Qt5::Svg>
            COMMENT "Compiling ${SVG_FILENAME} to ${QVG_FILENAME}")
    endfunction()
endif()
```

Reply:
The qvg files are "binaries" - compiled from svg sources using the svg2qvg tool, that is built with qskinny.
At the moment we only have SVG files in some examples, but soon there will also be SVG files being part of the material3 skin.

The idea of the svg2qvg tool is to do the conversion into QPainter commands at compile time. Note that the application driving the development of qskinny has more than 1000 icons, what makes such a precompilation worth.
It also avoids having to link the QtSvg module ( and its dependencies ) to qskinny.

However doing this compilation in the build stage causes problems for cross compilations as the svg2qvg tool has to run on the build system and then you would have an initial build for the host before actually starting the cross compilation.
As the qvg files do not depend on the architecture I decided to avoid the problem by generating the qvg files manually and then to commit them as sources. ( Something similar has been done for the qsb files in src/nodes: at least with Qt5 you can't rely on the qsb tool being available. ) Maybe the situation can be solved in a better way with cmake, but this is of no priority.

As the qmake rules for creating the qvg files were hardly won I didn't want to throw them away but for the cmake port you can simply ignore the autoqvg path.